src/main/scala/beam/utils/scenario/PreviousRunPlanMerger.scala

```diff
@@ -94,24 +124,50 @@ object PreviousRunPlanMerger extends LazyLogging {
           .zipWithIndex
           .take(maximumNumberOfPlansToKeep.getOrElse(0))
           .flatMap { case ((_, elems), idx) =>
-            elems.map { case elem =>
-              elem.copy(planIndex = idx + 1, planSelected = false)
-            }
+            elems.map(elem => elem.copy(planIndex = idx + 1, planSelected = false))
           }
       }
     }
-    oldElements ++ unselectedPlanElements ++ elementsFromExistingPersonsToAdd ++ elementsFromNewPersonsToAdd
+    oldElementsAfterChoosing ++ unselectedPlanElements ++ elementsFromExistingPersonsToAdd ++ elementsFromNewPersonsToAdd
+  }
+
+  private def drawFromGumbel(beta: Double, random: Random): Double = {
+    if (beta <= 0.0) { 0.0 }
+    else {
+      // CF https://en.wikipedia.org/wiki/Gumbel_distribution#Random_variate_generation
+      -beta * math.log(-math.log(random.nextDouble()))
+    }
   }
 }
 
 object LastRunOutputSource extends LazyLogging {
 
-  def findLastRunOutputPlans(outputPath: Path, dirPrefix: String): Option[Path] = {
+  def findLastRunOutputPlans(outputPath: Path, dirPrefix: String): (Option[Path], Option[Path]) = {
     val plansPaths = for {
       (itDir, itNumber) <- findAllLastIterationDirectories(outputPath, dirPrefix)
```

Comment:
The following 7 lines is the same as 157-164. Could we extract a method with a self-explanatory name?